contracts/dmarket.js

```diff
@@ -0,0 +1,1285 @@
+/* eslint-disable no-await-in-loop */
+/* global actions, api */
+
+const UTILITY_TOKEN_SYMBOL = 'BEE';
+
+// 30 days in seconds
+const MAX_EXPIRATION_SECS = 2592000;
+// fee in UTILITY TOKEN required to create a new pair
+const PAIR_CREATION_FEE = '500';
+
+const CONTRACT_NAME = 'dmarket';
+
+actions.createSSC = async () => {
+  const tableExists = await api.db.tableExists('pairs');
+
+  if (tableExists === false) {
+    await api.db.createTable('pairs', ['symbol', 'precision', 'allowedSymbols']);
```

Comment:
Please reduce the number of indexes, as they take up a lot of space. We should audit which queries you are using and optimize for that. Note that from recent times, we now have the ability to do compound indexes (meaning if you always are querying with pair/symbol while sorting by e.g. price, then you can make a compound index on [pair, symbol, priceDec]. For example. See https://github.com/hive-engine/steemsmartcontracts/blob/hive-engine/contracts/comments.js#L18 for examples